build-tools/build.sh

```diff
@@ -57,7 +57,7 @@ echo "::endgroup::"
 # ----------------
 echo "::group::Getting project info"
 
-sha=$GITHUB_SHA
+sha=$(echo $GITHUB_SHA | head -c 6)
```

Comment:
use the sha from git, GITHUB_SHA seems to not be what we want.
```bash
sha=$(git rev-parse HEAD | head -c 6)
```
[♻️ Rewrite for Docker #140](https://github.com/LemLib/pros-build/actions/runs/9426218742)
![image](https://github.com/LemLib/pros-build/assets/80860310/39a43089-e54f-4ab4-a67c-01c883abf815)

Reply:
For the test cases it's probably  not, but does that work if it's included in a user's repository (and in their Pull Requests)

Reply:
it might work in a user repo, but only if they're running the action on that repo. A user might run the action on a different repo as well.

Reply:
huh?

Reply:
if lemlib depends on a different template, we might also want to compile that template in our workflow maybe.
If we use GITHUB_SHA, we restrict the user to compiling only the repo the workflow is running on.

Reply:
The only way to presently do that is to overwrite the build script, since we only build the included template. (And that also doesn't make much sense since that isn't how it's done normally either). The only case where that's done is when you build the repo separately, and then copy the template to the user's re;po)

Reply:
> if lemlib depends on a different template, we might also want to compile that template in our workflow maybe.

How we could do this currently:
1. check out the dependency
2. run this action to build the dependency
3. store the dependency artifact somewhere
4. overwrite the dependency and check out lemlib
5. apply the dependency
6. build lemlib with this action

But that assumes that we retrieve the SHA from git and not `$GITHUB_SHA`.
`git rev-parse HEAD` does the same thing as `$GITHUB_SHA`, but is less limiting and doesn't introduce another dependency on github.

Reply:
You would basically have to do the following:
Job1:
Checkout the Template
Build the template (Our Workflow)
Upload that template
Job 2: (Wait for job 1)

Checkout the main repository
Build the main repository (Possibly not needed, but idk)
Download the template
Apply the template
Rebuild the main repository
Upload the artifact

In which case, GITHUB_SHA is still usable to my understanding. I'll switch to the `git rev-parse HEAD | head -c 6` command though

Reply:
GITHUB_SHA as I understand it refers to the sha of the event

Reply:
my evidence being ctrl+f `GITHUB_SHA` on [Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows)

Reply:
Ah, we'd have to make an env with `github.event.pull_request.head.sha` instead to get the same as git rev-parse HEAD